package tvserver;

message tune_param_vchannel {
    required uint32 version = 1;

    optional uint32 vchannel = 2;
}

message tune_param_channels_conf_record {
    required uint32 version = 1;

    optional string name = 2;
    optional uint32 frequency = 3;
    optional string modulation = 4;
    optional uint32 video_id = 5;
    optional uint32 audio_id = 6;
    optional uint32 program_id = 7;
}

message tune {
    required uint32 version = 1;
    optional string tuning_bigid = 2;

    enum parameter_type_e {
        VCHANNEL = 0;
        CHANNELSCONF = 1;
    }
    
    optional parameter_type_e parameter_type = 3;
    optional tune_param_vchannel vchannel = 4;
    optional tune_param_channels_conf_record channelsconf_record = 5;
}

